anomalib/integration/nncf/compression.py

```diff
@@ -0,0 +1,111 @@
+"""NNCF functions."""
+
+# Copyright (C) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import logging
+from typing import Any, Dict, Iterator, Tuple
+
+from nncf import NNCFConfig
+from nncf.torch import create_compressed_model, load_state, register_default_init_args
+from nncf.torch.compression_method_api import PTCompressionAlgorithmController
+from nncf.torch.initialization import PTInitializingDataLoader
+from nncf.torch.nncf_network import NNCFNetwork
+from torch import nn
+from torch.utils.data.dataloader import DataLoader
+
+logger = logging.getLogger(name="NNCF compression")
+logger.setLevel(logging.DEBUG)
+
+
+class InitLoader(PTInitializingDataLoader):
+    """Initializing data loader for NNCF to be used with unsupervised training algorithms."""
+
+    def __init__(self, data_loader: DataLoader):
+        super().__init__(data_loader)
+        self._data_loader_iter: Iterator
+
+    def __iter__(self):
+        """Create iterator for dataloader."""
+        self._data_loader_iter = iter(self._data_loader)
+        return self
+
+    def __next__(self) -> Any:
+        """Return next item from dataloader iterator."""
+        loaded_item = next(self._data_loader_iter)
+        return loaded_item["image"]
+
+    def get_inputs(self, dataloader_output) -> Tuple[Tuple, Dict]:
+        """Get input to model.
+
+        Returns:
+            (dataloader_output,), {}: Tuple[Tuple, Dict]: The current model call to be made during
+            the initialization process
+        """
+        return (dataloader_output,), {}
+
+    def get_target(self, _):
+        """Return structure for ground truth in loss criterion based on dataloader output.
+
+        This implementation does not do anything and is a placeholder.
+
+        Returns:
+            None
+        """
+        return None
```

Comment:
Is this a duplicate? I can see the same class defined in `anomalib/integration/nncf/utils.py`

Reply:
Looks like `InitLoader` is imported from `anomalib/integration/nncf/utils.py`. In this case, I feel like we could remove `anomalib/integration/nncf/compression.py`.

Reply:
The same class is also defined in `anomalib/utils/callbacks/nncf_callback.py`. Same for the `NNCFCallback` class itself.